core/contracts/financial-templates/implementation/PricelessPositionManager.sol

```diff
@@ -214,12 +215,13 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
         PositionData storage positionData = _getPositionData(msg.sender);
         require(positionData.requestPassTimestamp == 0);
 
-        _removeCollateral(positionData.rawCollateral, collateralAmount);
-        require(_checkPositionCollateralization(positionData));
+        // Decrement the caller's collateral and global collateral amounts.
+        amountWithdrawn = decrementCollateralBalances(positionData, collateralAmount);
+
         // We elect to withdraw the amount that the global collateral is decreased by,
         // rather than the individual position's collateral, because we need to maintain the invariant that
         // the global collateral is always <= the collateral owned by the contract to avoid reverts on withdrawals.
-        amountWithdrawn = _removeCollateral(rawTotalPositionCollateral, collateralAmount);
+        require(_checkPositionCollateralization(positionData));
```

Comment:
Note that this require is moved to after the position decrement to ensure that if the caller has decrease their balance such that it is no longer correctly collateralized the withdrawal will revert.